dashboard/test/models/pd/workshop_test.rb

```diff
@@ -492,6 +492,18 @@ class Pd::WorkshopTest < ActiveSupport::TestCase
     assert_equal 20, @workshop.effective_num_hours
   end
 
+  test 'time constraint lookup' do
+    workshop_bad_course = build :pd_workshop, course: 'nonexistent'
+    workshop_bad_subject = build :pd_workshop, course: Pd::Workshop::COURSE_CSP, subject: 'nonexistent'
+    workshop_ambiguous_subject_ecs = build :pd_workshop, course: Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_PHASE_2
```

Comment:
I don't really see how this is ambiguous? There's only one subject with that name

Reply:
The [subject text](https://github.com/code-dot-org/code-dot-org/blob/a3ca9a07c556497a6aae77b12ad02a170f112960/dashboard/app/models/pd/workshop.rb#L71) is identical:
```ruby
      SUBJECT_ECS_PHASE_2 = 'Phase 2 in-person'.freeze,
#...
      SUBJECT_CS_IN_A_PHASE_2 = 'Phase 2 in-person'.freeze,
```

Perhaps I should use the hard-coded string here so it's more obvious?
I'm not sure how useful this test case really is now that I changed the way we look these up.